### PR TITLE
TPC: Adding workflow for integrated DCAs for corrections and QC

### PR DESCRIPTION
**Pull request by matthias-kleiner**

### CODE REVIEW

File: Detectors/TPC/workflow/src/TPCIntegrateDCASpec.cxx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+      // DCAr
+      const float dcar = dca[0];
+
+      // weight of DCAr
+      const float resCl = track.getNClusters() / Mapper::PADROWS;
+      const float resCl2 = resCl * resCl;
+      const float fB = 1 / (resCl2 * track.getPt());
+
+      const float fA = 0.15;
+      const float dcarW = fA * fA + 0.2 * fB * fB; // Rms2 ~ 0.15^2 +  k/(L^2*pt) →    0.15**2 +  (0.2/((NCl/152)^2*pt)^2);
```
Comment (matthias-kleiner) [inline Detectors/TPC/workflow/src/TPCIntegrateDCASpec.cxx]: Hi @miranov25 ,
can you check if the formula for the weight is correct

Comment (miranov25) [reply Detectors/TPC/workflow/src/TPCIntegrateDCASpec.cxx]: Is there rescl2 squared? Should be only squared not Power 4

Comment (miranov25) [reply Detectors/TPC/workflow/src/TPCIntegrateDCASpec.cxx]: After discussion with @matthias-kleiner 
```
      const float resCl = track.getNClusters() / Mapper::PADROWS;
      const float fB = 0.2 / (resCl * track.getPt());

      const float fA = 0.15;
      const float dcarW = std::sqrt(fA * fA + fB * fB); // Rms2 ~ 0.15^2 +  k/(L^2*pt) →    0.15**2 +  (0.2/((NCl/152)^2*pt)^2);
```

**Comment by miranov25** [comment]
One more observation:
weight should be 1/rms not rms

**Comment by miranov25** [comment]
Hello @Mathias.

The information we get is very valuable in identifying the problems and understanding the detection system better. I expect that later we will add a new feature for monitoring.

I prefer to add 2 enhancements right now:
* Add qPt to the list of variables as we have sector and tgl now (20 bins, -5.5 )
* Add Ncl, MIP/dEdx, TPC/ITS chi2 as it is done now for DCAr, DCZ and TPC-ITS matching efficiency

Hopefully, we will get the device running in the data and MC production by default soon.

Regards
Marian

**Comment by miranov25** [comment]
I forgot, I would like to rename the debug tree and the summary tree to
* treeTimeSeries
The data we extract is not only about the distortion - DCAr, DCAz - but we also monitor other consequences of the imperfection of the detector, the reconstruction and the calibration process.

For example, deep in the TPC-ITS matching efficiency at the beginning of the run.

**Comment by miranov25** [comment]
* TPC/ITS chi2 - is defined only for the TPC ITS tracks
  * sqrt(track.getChi2Match())) 
* 50(MIP)/dEdx - both TPC and TPC+ITS
* TPC sqrt(chi2) - both TPC and TPC+ITS
* Ncl - both TPC and TPC+ITS

**Comment by matthias-kleiner** [comment]
> * TPC/ITS chi2 - is defined only for the TPC ITS tracks
>       
>       * sqrt(track.getChi2Match()))
> 
>     * 50(MIP)/dEdx - both TPC and TPC+ITS
> 
>     * TPC sqrt(chi2) - both TPC and TPC+ITS
> 
>     * Ncl - both TPC and TPC+ITS

I just added it. I will run now on the GSI for some TFs. Then we can check if everything is fine

**Comment by miranov25** [comment]
Hello @matthias-kleiner 

Thanks for the code and new data.  Look good. We can merge soon.

For the particular case of the TPC-ITS matching chi2 and matching efficiency, we should also put there 2 options - as 2 variables for the TPC+ITSstandalone  and for the  TPC+ITS afterburner.

I would like to disentangle these 2 cases both for the efficiency and matching chi2

image

**Comment by miranov25** [comment]
Hello @matthias-kleiner 

To ensure self-consistency, I propose to store the metadata in the TimeSeries tree. we have a mechanism to propagate it
In my trees in Run1,Run2 time series I have added the metadata as a dictionary to the UserInfo tree.
We need metadata with the axis title and a detailed description/definition of the variables. 

```python
df.meta.metaData = {
    "qpt.AxisTitle":"qpt (q/GeV)", "qpt.Description":"qpt (q/GeV)",
    "dTime.AxisTitle":"(t-t0) (s)", "qpt.Description":"Time  minus minimum time stamp in sample",
    "mITSTPC_A_MatchEff.AxisTitle":"eff_TPCITS (Unit)", 
    "mITSTPC_A_MatchEff.Description":"TPC-> ITS matching eff. |DCAr-median|<4 & |DCAz-median|<4",
    "mTSITSTPCmDCAr_A_Median.AxisTitle":"DCAr (cm)",
    "mTSITSTPCmDCAr_A_Median.Description":"DCAr (cm) for TPC tracks  requiring ITS and |DCAits+tpc|<0.4 ",
}
```

**Comment by miranov25** [comment]
Working well I checked the N dimensional dashboard- results looks consistent
image

**Comment by miranov25** [comment]
Hello @Matthias.

As we discussed in mattermost, to distinguish the performance of ITS-TPC standalone and ITS-TPC afterburner, we should have 2 versions of matching chi2 and matching efficiency.

1. For all matches, as you have already implemented (this does not have to change)
2. For ITS standalone - > TPC identified with the flag ```atch.getRefITS().getSource() == o2::dataformats::GlobalTrackID:: ITS ```

This is an important function because the primary and secondary tracks are affected differently, also the TPC-ITS matching chi2 is very different for these two groups of tracks.

Marian

**Comment by miranov25** [comment]
Hello @Mathias.
 I have seen your commit. We will need not only the efficiency but in the same way 
(All,TPCITSA,TPCITSAB)
but we ned also matching efficiency. This I did jot sort in your commit, maybe I missed it.

Marian

**Comment by mathias** [comment]
Hi @miranov25, I think you mean to @-mention @matthias-kleiner not me :)

**Comment by miranov25** [comment]
Hello @matthias-kleiner, @shahor02  and @ktf 

A test on the Mac has failed. The error message is not clear. We need to merge this ull request urgently as we need to enable the code in the upcoming reconstruction production for pp and Pbb.

Can you help @matthias-kleiner to solve the problem?

Thank you
Marian

**Comment by shahor02** [comment]
o2-cs8-perftest  and o2-dataflow-cs8-perftest are stuck, merging.

**Pull request merged**